integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/TableContentChangeDatasetBuilder.java

```diff
@@ -64,6 +65,8 @@ public List<OpenLineage.OutputDataset> apply(LogicalPlan x) {
       table = (NamedRelation) ((UpdateTable) x).table();
     } else if (x instanceof MergeIntoTable) {
       table = (NamedRelation) ((MergeIntoTable) x).targetTable();
+    } else if (x instanceof DataSourceV2ScanRelation) {
```

Comment:
I think those lines are not required. 
They're not executed due to `isDefinedAtLogicalPlan` method.